src/components/LandingPageMiddleware.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import { DEFAULT } from "../constants";
+import { useModulesManager } from "../helpers/modules";
+import Contributions from "./generics/Contributions";
+
+/**
+ * @name LandingPageMiddleware
+ * @param {boolean} props.isAuthenticated
+ * @returns {React.Component}
+ *
+ * @description
+ * This component is a middleware that checks if the user is authenticated and if the landing page is enabled.
+ * If the user is not authenticated, it redirects to the login page.
+ * If the user is authenticated, it redirects to the home page.
+ * If the landing page is not enabled, it redirects to the home page (if the user is authenticated, otherwise to the login page).
+ * If the user is not authenticated and the landing page is enabled, it renders the landing page.
+ * To enable the landing page, set the "App.enableLandingPage" configuration to true and
+ * add the "core.LandingPage" contribution from any module.
+ */
+
+const LANDING_PAGE_CONTRIBUTION_KEY = "core.LandingPage";
+
+const LandingPageMiddleware = ({ isAuthenticated }) => {
+  const modulesManager = useModulesManager();
+  const enableLandingPage = modulesManager.getConf("fe-core", "App.enableLandingPage", DEFAULT.ENABLE_LANDING_PAGE);
+
+  if (isAuthenticated || !enableLandingPage) {
```

Comment:
not sure if the `enableLandingPage` configuration is required in most cases. Perhaps we could do something like: 
```
if (isAuthenticated and noContributionsToLandingpage): 
   ... 
```
instead.

Reply:
My goal was to have more control over enabling or disabling this feature, rather than relying solely on the Contributions. This way, even without contributions, the public `/front` page would still function as expected, making the implementation (in my opinion) more straightforward. Let me know what do you think.

Reply:
That;s a valid point, but in that case please add info about the config into the README.md